bndtools.core/src/bndtools/central/Central.java

```diff
@@ -212,6 +215,20 @@ public synchronized static Workspace getWorkspace() throws Exception {
             newWorkspace.addBasicPlugin(new WorkspaceListener(newWorkspace));
             newWorkspace.addBasicPlugin(instance.repoListenerTracker);
             newWorkspace.addBasicPlugin(getWorkspaceR5Repository());
+
+            // need to remove any progress plugins that already exist so we can favor JobProgress
+            Set<Object> progressPlugins = new HashSet<>();
+
+            for (Object plugin : newWorkspace.getPlugins()) {
+                if (plugin instanceof ProgressPlugin) {
+                    progressPlugins.add(plugin);
+                }
+            }
+
+            for (Object plugin : progressPlugins) {
+                newWorkspace.removeBasicPlugin(plugin);
+            }
+
```

Comment:
Should I create a bnd issue for that?

On Wed, Jun 22, 2016 at 2:05 PM, BJ Hargrave <notifications@github.com>
wrote:

> In bndtools.core/src/bndtools/central/Central.java
> <https://github.com/bndtools/bndtools/pull/1393#discussion_r68115321>:
>
> > @@ -212,6 +215,20 @@ public synchronized static Workspace getWorkspace() throws Exception {
> >              newWorkspace.addBasicPlugin(new WorkspaceListener(newWorkspace));
> >              newWorkspace.addBasicPlugin(instance.repoListenerTracker);
> >              newWorkspace.addBasicPlugin(getWorkspaceR5Repository());
> > +
> > +            // need to remove any progress plugins that already exist so we can favor JobProgress
> > +            Set<Object> progressPlugins = new HashSet<>();
> > +
> > +            for (Object plugin : newWorkspace.getPlugins()) {
> > +                if (plugin instanceof ProgressPlugin) {
> > +                    progressPlugins.add(plugin);
> > +                }
> > +            }
> > +
> > +            for (Object plugin : progressPlugins) {
> > +                newWorkspace.removeBasicPlugin(plugin);
> > +            }
> > +
>
> Hmm, ok. Seems like it really should support multiple ProgressPlugins and
> not just one. But I guess this is ok for now.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/bndtools/bndtools/pull/1393/files/8c0bd9911f92fec87e4f0bea2b4d164121cfcf99..ef1896e7395a2dcf0e60ddb4e2300640816cd40d#r68115321>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAkVFcwOdCdbQwA01ORyj_qkAlBRtC5rks5qOYeDgaJpZM4IvtOy>
> .
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com